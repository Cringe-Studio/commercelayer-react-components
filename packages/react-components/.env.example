# Unit tests
VITE_TEST_CLIENT_ID=
VITE_TEST_ENDPOINT=
VITE_TEST_MARKET_ID=
VITE_TEST_USERNAME=
VITE_TEST_PASSWORD=
VITE_TEST_CLIENT_ID_INTEGRATION=
VITE_TEST_CLIENT_SECRET=
