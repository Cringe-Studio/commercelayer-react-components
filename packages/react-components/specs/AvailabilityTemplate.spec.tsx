import React from 'react'
import { AvailabilityTemplate } from '../packages/react-components/src'
import renderer from 'react-test-renderer'
import components from '../#config/components'

const propTypes = components.AvailabilityTemplate.propTypes

test('<AvailabilityTemplate/>', () => {
  expect.assertions(4)
  const component = renderer.create(<AvailabilityTemplate />)
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.timeFormat).toBe(propTypes.timeFormat)
  expect(proptypes.showShippingMethodName).toBe(
    propTypes.showShippingMethodName
  )
})

test('<AvailabilityTemplate with props />', () => {
  expect.assertions(3)
  const component = renderer.create(<AvailabilityTemplate />)
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.nodeType).toBe('host')
  expect(rendered.type).toBe('p')
})

test('<AvailabilityTemplate with custom children />', () => {
  expect.assertions(6)
  const CustomComponent = props => <span>{props.label}</span>
  const component = renderer.create(
    <AvailabilityTemplate timeFormat="hours">
      {CustomComponent}
    </AvailabilityTemplate>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const childRendered = root.rendered.rendered
  expect(tree).toMatchSnapshot()

  expect(rendered.props.children).toBe(CustomComponent)

  expect(childRendered.nodeType).toBe('component')
  expect(childRendered.type).toBe(CustomComponent)

  expect(childRendered.props.timeFormat).toBe('hours')
  expect(childRendered.props.showShippingMethodName).toBe(false)
})
