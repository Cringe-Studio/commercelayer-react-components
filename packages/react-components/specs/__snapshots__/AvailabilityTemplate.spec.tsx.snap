// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvailabilityTemplate with custom children /> 1`] = `<span />`;

exports[`<AvailabilityTemplate with props /> 1`] = `
<p>
  
</p>
`;

exports[`<AvailabilityTemplate/> 1`] = `
<p>
  
</p>
`;
